src/schema/v2/__tests__/fair.test.js

```diff
@@ -352,41 +350,6 @@ describe("Fair", () => {
         },
       },
     })
-
-    const nextQuery = gql`
-      {
-        fair(id: "aqua-art-miami-2018") {
-          shows: showsConnection(first: 1, after: "YXJyYXljb25uZWN0aW9uOjA=") {
-            pageInfo {
-              hasNextPage
-            }
-            edges {
-              node {
-                name
-              }
-            }
-          }
-        }
-      }
-    `
-    const nextData = await runQuery(nextQuery, context)
-
-    expect(nextData).toEqual({
-      fair: {
-        shows: {
-          pageInfo: {
-            hasNextPage: false,
-          },
-          edges: [
-            {
-              node: {
-                name: "B",
-              },
-            },
-          ],
-        },
-      },
-    })
```

Comment:
We spent some time looking at this spec, and frankly, we never quite understood a) how it was ever passing, and b) why we couldn't get it to pass. In its current form, the cursor passed to `after` is both the `endCursor` and the cursor for node `"A"`, which seems counterintuitive - why would an item that is _not_ the last item in the list have the same cursor as `endCursor`?

@anastasiapyzhik I think you mentioned that the `endCursor` might actually belong to the second-to-last item; maybe that's the behavior here and I'm just misunderstanding how `endCursor` works?

Reply:
One quote from the [relay spec](https://relay.dev/graphql/connections.htm):

> startCursor and endCursor must be the cursors corresponding to the first and last nodes in edges, respectively.

Reply:
`endCursor` is the cursor of the **last** edge of the **page**

<details>
<summary>When we fetch the very first 3 edges we <b>don't</b> pass any cursor and you can see that <code>endCursor</code> is equal to the cursor of the 3d edge</summary>

![Screenshot 2021-07-21 at 10 33 31](https://user-images.githubusercontent.com/44819355/126451328-87d5f23f-df92-4a96-a4f9-5a7bd6995130.png)

</details>

<details>
<summary>Then we use this <code>endCursor</code> as <code>after</code> (i.e. we fetch the next 3 edges <b>after</b> the 3d one)</summary>

![Screenshot 2021-07-21 at 10 34 54](https://user-images.githubusercontent.com/44819355/126451484-947819dd-e55f-4da6-894a-1e6846416ee4.png)

</details>

<details>
<summary>Big list of edges to ensure that pagination works correctly</summary>

![Screenshot 2021-07-21 at 10 33 06](https://user-images.githubusercontent.com/44819355/126450862-abf12b70-3a4a-4d42-a64c-2202a23e78a7.png)

</details>